aws_xray_sdk/core/models/entity.py

```diff
@@ -262,7 +262,7 @@ def serialize(self):
         try:
             return json.dumps(self.to_dict(), default=str)
         except Exception:
-            log.exception("Failed to serialize %s", self.name)
+            raise
```

Comment:
Same here. We don't need `try-except` clock if we are just raising exception. I think it would be better to log "Failed to serialize entity" message in the `except` and then raise the exception.

Reply:
I removed the `try-catch` block and don't think we need to log the message as the detailed stack trace will be printed out anyway